### PR TITLE
Networking Improvments (Drop unknown packets & don't hang if unable to connect to server)

### PR DESCRIPTION
**Pull request by AnthonyWharton**
Says what it does in the title. Please can this be checked on Windows by @qqii or @grant-m-s (or @jnterry?)

Fixes: #36, #100 

**Feel free to delete this branch once merged.**

### CODE REVIEW

**Comment by AnthonyWharton** [comment]
Actually also, probably worth testing on lab machines as I've only tested this locally..

**Comment by qqii** [comment]
Does this also close #98?

**Comment by qqii** [comment]
Since it is directly relevant, can we get an updated README.md with instructions about how to use it on a lab machine?

**Comment by qqii** [comment]
On a single lab machine here's what I've tested:

* Loading the client then the server allows the client to connect.
* Connecting two players is fine, there is latency but I am able to play the game from the other screen.
* Connecting a client to a server then crashing the server leave the client bouncing, restarting the server does not fix this.
* Repeatedly connecting the client and then disconnecting from the server leaves the client disconnected.

Camera FOV is something to investigate: #119

**Comment by qqii** [comment]
Instructions for Lab Machine:
1. Pull the repo and navigate to the root folder. Then run:
2. `mkdir build`
3. `cd build`
4. `cmake ..`
5. `make [-j8]`
6. `cd ../bin/`
7. Now you need your IP address, for that run `ifconfig` and look for the IPv4 address under `eno1`, it should be something like `137.222.xxx.xxx` ish...
8. Now run the server on one machine with `./RaveCave_Server`
9. Now run the client on any machine in the lab with `./RaveCave [ip_address]`
10. Et voila! You should be good to go :slightly_smiling_face:

**Comment by RcColes** [comment]
I'm unconvinced by having the readme change in here, as it's fundamentally a separate issue. It should have another github issue and PR created for it.

File: src/client/main.cpp [hunk 1]
```diff
@@ -211,9 +211,9 @@ int main(int argc, const char **argv)
 		tempo::addr_r = argv[1];
 	if (tempo::addr_r == "127.0.0.1") {
 		std::srand(time(NULL));
-		int d          = std::rand() % 10;
-		tempo::port_ci = DEFAULT_PORT_IN + 10 + d;
-		tempo::port_co = DEFAULT_PORT_OUT + 10 + d;
+		int d          = std::rand() % 1000;
```
Comment (RcColes) [inline src/client/main.cpp]: Technically an undocumented change, but I'm not complaining

Comment (AnthonyWharton) [reply src/client/main.cpp]: Snakes! This is for your benefit :stuck_out_tongue:

**Comment by AnthonyWharton** [comment]
Paging @qqii for updating review. The issue on the lab machines was not having the new ports merged in.

File: src/client/main.cpp (hunk 1 above)
Comment (qqii) [reply src/client/main.cpp]: This doesn't break anything does it? If it selects a port which is already being used for example.

Comment (AnthonyWharton) [reply src/client/main.cpp]: Then it will not be able to open sockets. But there's a reasonably low likelihood of that occurring.

**Pull request merged**